be/src/exec/pipeline/scan/scan_operator.cpp

```diff
@@ -100,31 +112,47 @@ bool ScanOperator::has_output() const {
     if (!_get_scan_status().ok()) {
         return true;
     }
-    if (has_buffer_output()) {
+
+    // Try to buffer enough chunks for exec thread, to reduce scheduling overhead.
+    // It's like the Linux Block-Scheduler's Unplug algorithm, so we just name it unplug.
+    // The default threshould of unpluging is BufferCapacity/DOP/4, and its range is [1, 16]
+    // The overall strategy:
+    // 1. If enough buffered chunks: pull_chunk, so return true
+    // 2. If not enough buffered chunks
+    //   2.1 Enough running io-tasks and buffer not full: wait some time for more chunks, so return false
+    //   2.1 Enough running io-tasks but buffer is full: pull_chunks, so return true
```

Comment:
Why under `not enough buffered chunks`, the buffer will be full, if so, could we get more chunks?

Reply:
Cause the `not enough` check the operator local buffer, but  the buffer full check the shared buffer.